protocol/vm/vm.go

```diff
@@ -11,10 +13,12 @@ import (
 const initialRunLimit = 10000
 
 type virtualMachine struct {
-	program      []byte
-	pc, nextPC   uint32
-	runLimit     int64
-	deferredCost int64
+	mainprog, curprog []byte
```

Comment:
Do we need to rename the existing field?

I'm not sure it's that much clearer to say curprog instead of program. But for the other one (which is just for the opcode right?) mainprog or outerProg or whatever definitely helps, and what might help even more is an end-of-line comment on these two that says why there are two of them. Roughly like

```
program  []byte // actual program currently executing
mainprog []byte // outermost program, returned by PROGRAM opcode
```